File: baselib/src/main/java/com/wan/baselib/network/interceptor/HeaderInterceptor.java
package com.wan.baselib.network.interceptor;
import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by Cy on 2017/5/26.
 * 添加请求头
 */
public class HeaderInterceptor implements Interceptor {

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();
        Request.Builder requestBuilder = originalRequest.newBuilder()
                .header("Content-Type", "application/json;charset=utf8")
                .header("Accept", "application/json")
                .method(originalRequest.method(), originalRequest.body());
        Request request = requestBuilder.build();
        return chain.proceed(request);
    }
}

File: baselib/src/main/java/com/wan/baselib/network/interceptor/HeaderInterceptor.kt
package com.wan.baselib.network.interceptor

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

/**
 * Created by Cy on 2017/5/26.
 * 添加请求头
 */
class HeaderInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest: Request = chain.request()
        val requestBuilder: Request.Builder = originalRequest.newBuilder()
            .header("Content-Type", "application/json;charset=utf8")
            .header("Accept", "application/json")
            .method(originalRequest.method, originalRequest.body)
        val request: Request = requestBuilder.build()
        return chain.proceed(request)
    }
}